File: app/src/main/java/com/gitaconnect/app/BhagavadGita DataStore/Verse.swift
import Foundation

struct ChapterVerse: Codable, Identifiable {
    var id: String { "\(chapter).\(verseId)" }

    let chapter: Int
    let verseId: Int
    let sanskritVerse: String
    let englishTranslation: String
    let essence: VerseEssence

    let hindiTranslation: String?
    let sanskritTranslation: String?
    let tamilTranslation: String?
    let teluguTranslation: String?
    let kannadaTranslation: String?
    let malayalamTranslation: String?
    let marathiTranslation: String?
    let gujaratiTranslation: String?
    let bengaliTranslation: String?
    let punjabiTranslation: String?
    let odiaTranslation: String?

    private enum CodingKeys: String, CodingKey {
        case chapter
        case verseId = "verse_id"
        case sanskritVerse = "sanskrit_verse"
        case englishTranslation = "english_translation"
        case essence

        case hindiTranslation = "hindi_translation"
        case sanskritTranslation = "sanskrit_translation"
        case tamilTranslation = "tamil_translation"
        case teluguTranslation = "telugu_translation"
        case kannadaTranslation = "kannada_translation"
        case malayalamTranslation = "malayalam_translation"
        case marathiTranslation = "marathi_translation"
        case gujaratiTranslation = "gujarati_translation"
        case bengaliTranslation = "bengali_translation"
        case punjabiTranslation = "punjabi_translation"
        case odiaTranslation = "odia_translation"
    }
    var verse: Int { verseId }
}

struct VerseEssence: Codable {
    let krishnaGuidance: String
    let reflection: String

    let krishnaGuidanceEnglish: String?
    let krishnaGuidanceHindi: String?
    let krishnaGuidanceSanskrit: String?
    let krishnaGuidanceTamil: String?
    let krishnaGuidanceTelugu: String?
    let krishnaGuidanceKannada: String?
    let krishnaGuidanceMalayalam: String?
    let krishnaGuidanceMarathi: String?
    let krishnaGuidanceGujarati: String?
    let krishnaGuidanceBengali: String?
    let krishnaGuidancePunjabi: String?
    let krishnaGuidanceOdia: String?

    let reflectionEnglish: String?
    let reflectionHindi: String?
    let reflectionSanskrit: String?
    let reflectionTamil: String?
    let reflectionTelugu: String?
    let reflectionKannada: String?
    let reflectionMalayalam: String?
    let reflectionMarathi: String?
    let reflectionGujarati: String?
    let reflectionBengali: String?
    let reflectionPunjabi: String?
    let reflectionOdia: String?

    private enum CodingKeys: String, CodingKey {
        case krishnaGuidance = "krishna_guidance"
        case reflection

        case krishnaGuidanceEnglish = "krishna_guidance_english"
        case krishnaGuidanceHindi = "krishna_guidance_hindi"
        case krishnaGuidanceSanskrit = "krishna_guidance_sanskrit"
        case krishnaGuidanceTamil = "krishna_guidance_tamil"
        case krishnaGuidanceTelugu = "krishna_guidance_telugu"
        case krishnaGuidanceKannada = "krishna_guidance_kannada"
        case krishnaGuidanceMalayalam = "krishna_guidance_malayalam"
        case krishnaGuidanceMarathi = "krishna_guidance_marathi"
        case krishnaGuidanceGujarati = "krishna_guidance_gujarati"
        case krishnaGuidanceBengali = "krishna_guidance_bengali"
        case krishnaGuidancePunjabi = "krishna_guidance_punjabi"
        case krishnaGuidanceOdia = "krishna_guidance_odia"

        case reflectionEnglish = "reflection_english"
        case reflectionHindi = "reflection_hindi"
        case reflectionSanskrit = "reflection_sanskrit"
        case reflectionTamil = "reflection_tamil"
        case reflectionTelugu = "reflection_telugu"
        case reflectionKannada = "reflection_kannada"
        case reflectionMalayalam = "reflection_malayalam"
        case reflectionMarathi = "reflection_marathi"
        case reflectionGujarati = "reflection_gujarati"
        case reflectionBengali = "reflection_bengali"
        case reflectionPunjabi = "reflection_punjabi"
        case reflectionOdia = "reflection_odia"
    }
}

File: app/src/main/java/com/gitaconnect/app/library/models/Verse.kt
package com.gitaconnect.app.library.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents a single verse, mapping from Chapter_N.json.
 * Equivalent to the Swift Verse model.
 */
@Serializable
data class Verse(
    val chapter: Int,
    @SerialName("verse_id") val verseId: Int,
    @SerialName("sanskrit_verse") val sanskritVerse: String,
    @SerialName("english_translation") val englishTranslation: String,
    val essence: Essence? = null,
    
    @SerialName("hindi_translation") val hindiTranslation: String? = null,
    @SerialName("sanskrit_translation") val sanskritTranslation: String? = null,
    @SerialName("tamil_translation") val tamilTranslation: String? = null,
    @SerialName("telugu_translation") val teluguTranslation: String? = null,
    @SerialName("kannada_translation") val kannadaTranslation: String? = null,
    @SerialName("malayalam_translation") val malayalamTranslation: String? = null,
    @SerialName("marathi_translation") val marathiTranslation: String? = null,
    @SerialName("gujarati_translation") val gujaratiTranslation: String? = null,
    @SerialName("bengali_translation") val bengaliTranslation: String? = null,
    @SerialName("punjabi_translation") val punjabiTranslation: String? = null,
    @SerialName("odia_translation") val odiaTranslation: String? = null
)

/**
 * The reflective guidance embedded in each verse's "essence" object.
 */
@Serializable
data class Essence(
    @SerialName("krishna_guidance") val krishnaGuidance: String? = null,
    val reflection: String? = null,

    @SerialName("krishna_guidance_english") val krishnaGuidanceEnglish: String? = null,
    @SerialName("krishna_guidance_hindi") val krishnaGuidanceHindi: String? = null,
    @SerialName("krishna_guidance_sanskrit") val krishnaGuidanceSanskrit: String? = null,
    @SerialName("krishna_guidance_tamil") val krishnaGuidanceTamil: String? = null,
    @SerialName("krishna_guidance_telugu") val krishnaGuidanceTelugu: String? = null,
    @SerialName("krishna_guidance_kannada") val krishnaGuidanceKannada: String? = null,
    @SerialName("krishna_guidance_malayalam") val krishnaGuidanceMalayalam: String? = null,
    @SerialName("krishna_guidance_marathi") val krishnaGuidanceMarathi: String? = null,
    @SerialName("krishna_guidance_gujarati") val krishnaGuidanceGujarati: String? = null,
    @SerialName("krishna_guidance_bengali") val krishnaGuidanceBengali: String? = null,
    @SerialName("krishna_guidance_punjabi") val krishnaGuidancePunjabi: String? = null,
    @SerialName("krishna_guidance_odia") val krishnaGuidanceOdia: String? = null,

    @SerialName("reflection_english") val reflectionEnglish: String? = null,
    @SerialName("reflection_hindi") val reflectionHindi: String? = null,
    @SerialName("reflection_sanskrit") val reflectionSanskrit: String? = null,
    @SerialName("reflection_tamil") val reflectionTamil: String? = null,
    @SerialName("reflection_telugu") val reflectionTelugu: String? = null,
    @SerialName("reflection_kannada") val reflectionKannada: String? = null,
    @SerialName("reflection_malayalam") val reflectionMalayalam: String? = null,
    @SerialName("reflection_marathi") val reflectionMarathi: String? = null,
    @SerialName("reflection_gujarati") val reflectionGujarati: String? = null,
    @SerialName("reflection_bengali") val reflectionBengali: String? = null,
    @SerialName("reflection_punjabi") val reflectionPunjabi: String? = null,
    @SerialName("reflection_odia") val reflectionOdia: String? = null
)
